### DISCUSSION TITLE
How to get the sum of a modified array measure?

### DISCUSSION THREAD

**Discussion by eric-kaufmann** [atoti/atoti#636] [Q&A]
Hi,

I have the following table 

 *id* | date_of_change | array
-- | -- | --
a | 2023-03-31 | [1.0, 2.0, 3.0, 4.0, 5.0]
b | 2023-01-31 | [1.5, 2.5, 3.5, 4.5, 5.5]
c | 2022-12-31 | [0.7, 1.7, 2.7, 3.7, 4.7]

where the values of the array are linked to a specific date in *daterange*
```python
print(daterange.to_list())

[Timestamp('2022-12-31 00:00:00', freq='M'),
 Timestamp('2023-01-31 00:00:00', freq='M'),
 Timestamp('2023-02-28 00:00:00', freq='M'),
 Timestamp('2023-03-31 00:00:00', freq='M'),
 Timestamp('2023-04-30 00:00:00', freq='M')]
```

So I create a parameter hierarchy

```python
cube.create_parameter_hierarchy_from_members('dates', daterange.to_list(), index_measure_name='dates_index')
```
Now I wan't to calculate the sum of all values from the array where *dates* > *date_of_change*

So the solution would look like this

id | date_of_change | new_array | sum
-- | -- | -- | --
a | 2023-03-31 | [0.0, 0.0, 0.0, 0.0, 5.0] | 5.0
b | 2023-01-31 | [0.0, 0.0, 3.5, 4.5, 5.5] | 13.5
c | 2022-12-31 | [0.0, 1.7, 2.7, 3.7, 4.7] | 12.8

Getting *new_array* isn't the problem. It can be calulated by

```python
m['new_array'] = tt.where(
    l['dates'] > l['date_of_change'],
    m['array_date_view'],
    0
)
```
or
```python
m['new_array'] = tt.where(
    l['dates'] > l['date_of_change'],
    m['array.SUM'],
    m['array.SUM'] * 0
)
```

but how do I get the *sum*? `tt.agg.sum` and `tt.array.sum()` doesn't work.

Thanks in advance :)

**Comment by Skyscimitar** [comment]
Hello,

When you create a hierarchy with `cube.create_parameter_hierarchy_from_members` (or with `cube.create_parameter_hierarchy_from_column`) it is ***slicing*** by default. If you want to see the be able to aggregate on this hierarchy you must make it non-slicing: `h["dates"].slicing = False`

In your use case, I would recommend using the `index_measure` to explode the array, apply the `where` function to the resulting measure, and then aggregate that over the `dates` level.

```python
h["date"].slicing = False
m["array_date_view"] = tt.value(table["array"])[m["dates_index"]]
values_after_change = tt.where(l["dates"] > l["date_of_change"], m["array_date_view"], 0)
m["values_after_change.SUM"] = tt.agg.sum(values_after_change, scope=tt.scope.origin(l["dates"]))
```

Which will give you the following result:
![date_sum](https://user-images.githubusercontent.com/24211435/178305729-2792593d-51cd-4163-9dbf-f103dd6a079f.png)
